tests/test_base_models.py

```diff
@@ -915,19 +916,159 @@ def event_state(self, x):
+        # when matrix is not of type NumPy ndarray or standard list
+        # @A
+        with self.assertRaises(TypeError):
+            m.A = "[[0, 1], [0, 0]]" # string
+            m.matrixCheck() 
+        with self.assertRaises(TypeError):
+            m.A = None # None
+            m.matrixCheck() 
+        with self.assertRaises(TypeError):
+            m.A = 0 # int
+            m.matrixCheck()
+        with self.assertRaises(TypeError):
+            m.A = 3.14 # float
+            m.matrixCheck()
+        with self.assertRaises(TypeError):
+            m.A = {} # dict
+            m.matrixCheck() 
+        with self.assertRaises(TypeError):
+            m.A = () # tuple
+            m.matrixCheck() 
+        with self.assertRaises(TypeError):
+            m.A = set() # set
+            m.matrixCheck() 
+        with self.assertRaises(TypeError):
+            m.A = true # boolean
+            m.matrixCheck()
+        # @C
+        with self.assertRaises(TypeError):
+            m.C = "[[0, 1], [0, 0]]" # string
+            m.matrixCheck() 
+        with self.assertRaises(TypeError):
+            m.C = None # None
+            m.matrixCheck() 
+        with self.assertRaises(TypeError):
+            m.C = 0 # int
+            m.matrixCheck()
+        with self.assertRaises(TypeError):
+            m.C = 3.14 # float
+            m.matrixCheck()
+        with self.assertRaises(TypeError):
+            m.C = {} # dict
+            m.matrixCheck() 
+        with self.assertRaises(TypeError):
+            m.C = () # tuple
+            m.matrixCheck() 
+        with self.assertRaises(TypeError):
+            m.C = set() # set
+            m.matrixCheck() 
+        with self.assertRaises(TypeError):
+            m.C = true # boolean
+            m.matrixCheck()
+        # @E
+        with self.assertRaises(TypeError):
+            m.E = "[[0, 1], [0, 0]]" # string
+            m.matrixCheck() 
+        with self.assertRaises(TypeError):
+            m.E = None # None
+            m.matrixCheck() 
+        with self.assertRaises(TypeError):
+            m.E = 0 # int
+            m.matrixCheck()
+        with self.assertRaises(TypeError):
+            m.E = 3.14 # float
+            m.matrixCheck()
+        with self.assertRaises(TypeError):
+            m.E = {} # dict
+            m.matrixCheck() 
+        with self.assertRaises(TypeError):
+            m.E = () # tuple
+            m.matrixCheck() 
+        with self.assertRaises(TypeError):
+            m.E = set() # set
+            m.matrixCheck() 
+        with self.assertRaises(TypeError):
+            m.E = true # boolean
+            m.matrixCheck()
+        # @F
+        with self.assertRaises(TypeError):
+            m.F = "[[0, 1], [0, 0]]" # string
+            m.matrixCheck() 
         with self.assertRaises(TypeError):
-            m.A = [[0, 1], [0, 0]] # standard 2-D array
+            m.F = None # None
```

Comment:
F can be none (if users specify an event_state function). This shouldn't raise a TypeError